src/NuGet.Clients/NuGet.PackageManagement.UI/Models/Package/VulnerableCapability.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Protocol;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.UI
+{
+    public class VulnerableCapability : IVulnerable
+    {
+        private IEnumerable<PackageVulnerabilityMetadataContextInfo> _vulnerabilities;
+
+        public IEnumerable<PackageVulnerabilityMetadataContextInfo> Vulnerabilities
+        {
+            get => _vulnerabilities;
+            private set => _vulnerabilities = value.OrderByDescending(v => v?.Severity ?? -1);
+        }
+
+        public bool IsVulnerable => Vulnerabilities.Any(v => v != null);
+
+        public PackageVulnerabilitySeverity VulnerabilityMaxSeverity
+        {
+            get
+            {
+                // Vulnerabilities are ordered so the first element is always the highest severity
```

Comment:
This is great for performance, not sure if we are currently iterating. More context: https://learn.microsoft.com/en-us/nuget/api/vulnerability-info#vulnerability-page.

Reply:
Correct, there's no way to guarantee the list will be ordered from any and all vulnerability sources, both current and future. This comment points out that the list is ordered. It's ordered in this class (see the setter) explicitly to avoid making assumptions about the data source. Does that help? Do you have any suggested edits?

Reply:
It does, right? `The list of known vulnerabilities for a package should be sorted in descending order of the max version of the version range`.

Reply:
Doesn't that talk about the way the version ranges are sorted and not the list of vulnerabilities?